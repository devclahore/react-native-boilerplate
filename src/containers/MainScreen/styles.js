/*
  Author: Taimoor Khan
  GitHub: https://github.com/Taimoormk
  Email: taimoor.m.k AT LIVE.COM 
*/

// ########## Import Dependencies Here ##########
import { StyleSheet } from "react-native";

import { STYLE_CONSTANTS, width } from "../../config/styles";

const styles = StyleSheet.create({

});

export default styles;
