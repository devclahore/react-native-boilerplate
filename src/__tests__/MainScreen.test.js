/*
  Author: Taimoor Khan
  GitHub: https://github.com/Taimoormk
  Email: taimoor.m.k AT LIVE.COM 
*/

// ########## Import Dependencies Here ##########
import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import configureStore from "redux-mock-store";

// ########## Import Components Here ##########
import MainScreen from "../containers/MainScreen";

// ########## Create Initial State ##########
const fakeInitialState = {
  showAppReducer: {
    showApp: true,
  }
};

const mockStore = configureStore();

test("renders MainScreen compomenent correctly", () => {
  const renderer = new ShallowRenderer();
  const showApp = false;
  const store = mockStore(fakeInitialState);
  const showAppAction = () => null;
  renderer.render(<MainScreen store={store} showApp={showApp} showAppAction={showAppAction} />);
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
