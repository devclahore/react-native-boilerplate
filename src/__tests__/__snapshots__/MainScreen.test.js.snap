// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainScreen compomenent correctly 1`] = `
<Context.Provider
  value={null}
>
  <MainScreen
    showApp={true}
    showAppAction={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</Context.Provider>
`;
